cnd/src/network/comit_ln.rs

```diff
@@ -105,30 +105,31 @@ impl ComitLN {
         &mut self,
         id: LocalSwapId,
         create_swap_params: HanEtherereumHalightBitcoinCreateSwapParams,
-    ) {
+    ) -> anyhow::Result<()> {
         let digest = create_swap_params.clone().digest();
 
+        if self.swaps_waiting_for_announcement.contains_key(&digest) {
+            return Err(anyhow::anyhow!(
+                "Swap with same digest already exists: {}",
+                digest
+            ));
+        }
         self.swaps.insert(id, create_swap_params.clone());
+        // No race condition because we are called within a mutex guard.
```

Comment:
This is guaranteed by the mutable reference at compile-time, not the Mutex guard!

Long live Rust!

Reply:
Long live Rust